src/gui/Layout.cpp

```diff
@@ -178,176 +179,150 @@ double Layout::getLayoutWidth()
  */
 const int XOURNAL_PADDING = 10;
 
+/**
+ * Allowance for shadow between page pairs in dual mode
+ */
+const int XOURNAL_ROOM_FOR_SHADOW = 3;
+
 /**
  * Padding between the pages
  */
 const int XOURNAL_PADDING_BETWEEN = 15;
 
+	
+
 void Layout::layoutPages()
 {
 	XOJ_CHECK_TYPE(Layout);
 
-	int y = 0;
-
 	int len = this->view->viewPagesLen;
 
 	Settings* settings = this->view->getControl()->getSettings();
-	bool verticalSpace = settings->getAddVerticalSpace();
-	bool horizontalSpace = settings->getAddHorizontalSpace();
+	bool verticalSpace = settings->getAddVerticalSpace();				//TODO: Use these again
+	bool horizontalSpace = settings->getAddHorizontalSpace();		//TODO: Use these again
 	bool dualPage = settings->isShowTwoPages();
-
-	int size[2] = { 0, 0 };
-
-	// we need at least 2 page for dual page view
-	if (len < 2)
-	{
-		dualPage = false;
+	int numRowsOrColumns = settings->getViewColumns();
+	int pairsOffset = settings->getPairsOffset();
+	bool vert = settings->getViewLayoutVert();
+	bool r2l = settings->getViewLayoutR2L();
+	bool b2t = settings->getViewLayoutB2T();
+	
+	int pagesOffset = 0;
+	
+	if (!dualPage ) pairsOffset = 0;
+	
+	LayoutMapper mapper( len, numRowsOrColumns, numRowsOrColumns, false, vert, r2l, b2t , dualPage, pairsOffset );
```

Comment:
Just add the `settings`object as argument and read all the flags in the LayoutMapper.

Reply:
I think of the layoutMapper as a simple Math type object. It's pretty lightweight. Many of the Settings properties are used where it's called as well.
I plan to add a menu for switching to fixed rows ( instead of columns) and I'd like to try this change once that's been done.
Also.. the AddVerticalSpace and HorizontalSpace settings need to be used here.. although I'm not sure as to their reason for existence?

Reply:
Still it is not very nice to manually grab all these variables each time we instanciate a Layoutmapper. It's smother to just provide the settings and the Layoutmapper gets everything it needs. If we change anything later on we would have to check each position where it is instantiated instead of changing it once in the constructor.